localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -569,7 +572,7 @@ bool NDTScanMatcher::callback_sensor_points_main(
 
   if (is_converged && param_.covariance.covariance_estimation.enable) {
     const auto estimated_covariance =
-      estimate_covariance(ndt_result, initial_pose_matrix, sensor_ros_time);
+      estimate_covariance(ndt_result, initial_pose_matrix, sensor_ros_time, ndt_covariance);
```

Comment:
It seems a bit odd that the function `estimate_covariance` takes `ndt_covariance` as an argument.
I think the covariance should be the return value.

Reply:
Now that you mention it, I think that's certainly true.
It may be strange to use `ndt_covariance` as an argument to output `estimate_covariance`.
However, within the estimate_covariance(), we need the default covariance (z, roll, pitch, yaw) converted into the base_link coordinate system.

Another option is to have the function return only the covariance of x and y, and not use `ndt_covariance` as an argument. However, there is a concern that the code will become a little longer from line 573 onwards. 
Should I include `ndt_covariance` as an argument, or not?

I also think the easiest way to avoid confusion would be to rename `ndt_covariance` to `default_ndt_covariance`.

Reply:
I think `estimate_covariance` should return not `std::array<double, 36>` but `Eigen::Matrix2d`. It is good to use `pclomp::adjust_diagonal_covariance` outside of `estimate_covariance`.

Reply:
I agree with your opinion.
I change the return value of `estimate_covariance()` to `Eigen::Matrix2d`.